code-challenges/fizzBuzzTree/fizz-buzz-tree.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+class Node {
+  constructor(value = null, left = null, right = null) {
+    this.value = value;
+    this.left = left;
+    this.right = right;
+  }
+}
+
+function fizzBuzzTree(treeA) {
+  const rootA = treeA.root;
+  const treeB = new Tree(new Node);
+  const rootB = treeB.root;
+  function fizzBuzz(rootA, rootB) {
+    if (rootA.value % 3 === 0 && rootA.value % 5 === 0) {
+      rootB.value = 'FizzBuzz';
+    } else if (rootA.value % 3 === 0) {
+      rootB.value = 'Fizz';
+    } else if (rootA.value % 5 === 0) {
+      rootB.value = 'Buzz';
+    } else {
+      rootB.value = rootA.value.toString();
+    }
+  }
+  function traverse(rootA, rootB) {
+    fizzBuzz(rootA, rootB);
+    if (rootA.left !== null) {
+      rootB.left = new Node();
+      traverse(rootA.left, rootB.left)
+    }
+    if (rootA.right !== null) {
+      rootB.right = new Node();
+      traverse(rootA.right, rootB.right)
+    }
+  }
+  traverse(rootA, rootB);
+  return treeB;
+}

```

Comment:
Excellent, your code generates a new fizzbuzz tree which is exactly the most challenging part of this code challenge.